### ISSUE TITLE
Allow to disable and grey out choices in in select and multiselect prompt types

### ISSUE THREAD

**Issue by pvdlg** [terkelg/prompts#96]
When using the [select](https://github.com/terkelg/prompts#selectmessage-choices-initial) or [multiselct](https://github.com/terkelg/prompts#multiselectmessage-choices-initial-max-hint) it might be useful to option some of the options being disabled and greyed out in order to communicate to the user that those choices exists but are not available for a certain reason.

For example in a CLI that allow to create, edit and delete entries in some data storage, the edit and delete options make senses only if there is existing entries. You might want to prompt the user with a choice like this:
```bash
? Please select mode
  Create
  Edit
  Delete
```

Currently, if there is no entries in the storage possible solution would be to:
- Skip the mode selection and directly go to creation, but that would be confusing for the user and fail to communicate that entries can later be edited or deleted
- Use the `onSubmit` callback to display a message explaining there is currently no entry so one has to be created first but that's not a great user experience.

We could add a `disabled` and a `hint` props to the choices object. The select prompt could be configured like this:
```
{
    type: 'select',
    name: 'mode',
    message: 'Please select mode',
    choices: [
        { title: 'Create', value: 'create' },
        {
          title: 'Edit',
          value: 'edit',
          enabled: () => entries.length > 0,
          hint: () => entries.length > 0 ? 'There is currently no entries in DB' : null
        },
        {
          title: 'Delete',
          value: 'delete',
          enabled: () => entries.length > 0,
          hint: () => entries.length > 0 ? 'There is currently no entries in DB' : null
        },
    ],
    initial: 1
}
```

And the prompt would look like this:
```bash
? Please select mode
  Create
  Edit - There is currently no entries in DB
  Delete - There is currently no entries in DB
```
with the Edit and Delete lines being greyed out.

**Comment by terkelg** [comment]
Sounds good to me. I'm open to this if you do a PR 👍

**Comment by pvdlg** [comment]
#88 is about passing the previous answer. This one is more the ability to disable and grey out option.
But we can definitely include both in the same PR as it's related (we could determine the disabled status based on the previous answers).

I don't think I'll have much time to work on that in the next few weeks/months. So if anyone want to work on that. feel free to do so. Otherwise I'll get to it at some point but don't know when.

**Comment by terkelg** [comment]
Added in `master` with #102 by @DrunkenPoney

**Comment by terkelg** [comment]
Published in [v1.2.0](https://github.com/terkelg/prompts/releases/tag/v1.2.0). Thanks!
